src/git_config/git_config_entry.rs

```diff
@@ -13,8 +13,31 @@ pub enum GitConfigEntry {
 }
 
 #[derive(Clone, Debug, PartialEq)]
-pub enum GitRemoteRepo {
-    GitHubRepo(String),
+pub struct GitRemoteRepo {
+    repo_slug: String,
+    commit_link_format: String,
+}
+
+impl GitRemoteRepo {
+    pub fn github_repo(repo_slug: String) -> Self {
+        Self {
+            repo_slug,
+            commit_link_format: "https://github.com/{repo_slug}/commit/{commit}".to_string(),
+        }
+    }
+
+    pub fn gitlab_repo(repo_slug: String) -> Self {
+        Self {
+            repo_slug,
+            commit_link_format: "https://gitlab.com/{repo_slug}/-/commit/{commit}".to_string(),
+        }
+    }
+
+    pub fn format_commit_url(&self, commit: &str) -> String {
+        self.commit_link_format
+            .replace("{repo_slug}", &self.repo_slug)
+            .replace("{commit}", commit)
+    }
```

Comment:
Hi @jwarlander, thanks very much for this PR!

Sorry for being slow. I'd been caught between wanting to leave a comment to keep things moving and wanting to find time not to leave a lazy comment, if you know what I mean.

My one question about this is whether perhaps the implementation isn't quite the idiomatic way to do this in Rust. 
 Since a repo cannot simultaneously be a GitHub repo and a GitLab repo, I was thinking that the way we'd implement this would be to add a `GitLabRepo` variant to the `GitRemoteRepo` enum. Maybe you've already thought about this while you were implementing this PR. What do you think, could it maybe be something like the code fragment below?

```rust
#[derive(Clone, Debug, PartialEq)]
pub enum GitRemoteRepo {
    GitHubRepo(String),
    GitLabRepo(String),
}

impl GitRemoteRepo {
    pub fn format_commit_url(&self, commit: &str) -> String {
        match self {
            Self::GitHubRepo(repo_slug) => {
                format!("https://github.com/{}/commit/{}", repo_slug, commit)
            }
            Self::GitLabRepo(repo_slug) => {
                format!("https://gitlab.com/{}/-/commit/{}", repo_slug, commit)
            }
        }
    }
}

fn format_commit_line_captures_with_osc8_commit_hyperlink(
    captures: &Captures,
    repo: &GitRemoteRepo,
) -> String {
    let commit = captures.get(2).unwrap().as_str();
    format!(
        "{prefix}{osc}8;;{url}{st}{commit}{osc}8;;{st}{suffix}",
        url = repo.format_commit_url(commit),
        commit = commit,
        prefix = captures.get(1).map(|m| m.as_str()).unwrap_or(""),
        suffix = captures.get(3).unwrap().as_str(),
        osc = "\x1b]",
        st = "\x1b\\"
    )
}

```

There might be better ways to do this; for example it would be nice if `GitHubRepo(String)` were more self-documenting regarding what the string data is.

Reply:
No worries, I definitely know how it goes, and I imagine you've got plenty to do besides working on this project :sweat_smile: 

Anyhow, yeah, that works of course!

It just didn't occur to me at the time of writing that you could actually have an impl for an enum type -- haven't had the opportunity to work with those a lot yet, and they're significantly more capable in Rust than other languages I've used :sunglasses:

Reply:
It would definitely be more self-documenting with the variants as structs, even if *slightly* more verbose; have a look at my most recent update to see how it would look.